src/utilities/texture_atlas.jl

```diff
@@ -396,86 +439,86 @@ primitive_offset(x, scale) = scale ./ -2f0  # default offset
 """
 Extracts the uv offset and width from a primitive.
 """
-primitive_uv_offset_width(x) = Vec4f(0,0,1,1)
-primitive_uv_offset_width(b::Union{Char, BezierPath}) = glyph_uv_width!(b)
-primitive_uv_offset_width(x::AbstractVector) = map(primitive_uv_offset_width, x)
-function primitive_uv_offset_width(marker::Observable)
-    return lift(primitive_uv_offset_width, marker; ignore_equal_values=true)
+primitive_uv_offset_width(atlas::TextureAtlas, x, font) = Vec4f(0,0,1,1)
+primitive_uv_offset_width(atlas::TextureAtlas, b::BezierPath, font) = glyph_uv_width!(atlas, b)
+primitive_uv_offset_width(atlas::TextureAtlas, b::Union{UInt64, Char}, font) = glyph_uv_width!(atlas, b, font)
+primitive_uv_offset_width(atlas::TextureAtlas, x::AbstractVector, font) = map(m-> primitive_uv_offset_width(atlas, m, font), x)
+function primitive_uv_offset_width(atlas::TextureAtlas, marker::Observable, font::Observable)
+    return lift((m, f)-> primitive_uv_offset_width(atlas, m, f), marker, font; ignore_equal_values=true)
 end
 
 _bcast(x::Vec) = (x,)
 _bcast(x) = x
 
 # Calculates the scaling factor from unpadded size -> padded size
-# Here we assume the glyph to be representative of Makie.PIXELSIZE_IN_ATLAS[]
+# Here we assume the glyph to be representative of pix_per_glyph
 # regardless of its true size.
-function marker_scale_factor(char::Char, font)
-    ta = Makie.get_texture_atlas()
-    lbrt = glyph_uv_width!(ta, char, font)
+function marker_scale_factor(atlas::TextureAtlas, char::Char, font)
+    lbrt = glyph_uv_width!(atlas, char, font)
     uv_width = Vec(lbrt[3] - lbrt[1], lbrt[4] - lbrt[2])
-    full_pixel_size_in_atlas = uv_width .* Vec2f(size(ta.data) .- 1)
-    return full_pixel_size_in_atlas ./ Makie.PIXELSIZE_IN_ATLAS[]
+    full_pixel_size_in_atlas = uv_width .* Vec2f(size(atlas) .- 1)
```

Comment:
This should be `size(atlas)` now as well, I think. Same with other divisions or multiplications by `size(atlas) - 1)`.